test/extensions/filters/udp/udp_proxy/udp_proxy_filter_test.cc

```diff
@@ -480,6 +590,147 @@ cluster: fake_cluster
   EXPECT_EQ(1, config_->stats().downstream_sess_active_.value());
 }
 
+// Make sure socket option is set correctly if use_original_src_ip is set.
+TEST_F(UdpProxyFilterTest, SocketOptionForUseOriginalSrcIp) {
+  if (!isTransparentSocketOptionsSupported()) {
+    // The option is not supported on this platform. Just skip the test.
+    GTEST_SKIP();
+  }
+
+  InSequence s;
+
+  EXPECT_CALL(os_sys_calls_, supportsIpTransparent());
+
+  setup(R"EOF(
+stat_prefix: foo
+cluster: fake_cluster
+use_original_src_ip: true
+)EOF");
+
+  expectSessionCreate(upstream_address_);
+  test_sessions_[0].expectUpstreamWriteOriginalSrcIp();
+  test_sessions_[0].expectUpstreamWrite("hello", 0, peer_address_->ip());
+  recvDataFromDownstream(peer_address_->asString(), "10.0.0.2:80", "hello");
+
+  checkSocketOptions(test_sessions_[0], ENVOY_SOCKET_IP_TRANSPARENT, 1,
+                     ENVOY_SOCKET_IPV6_TRANSPARENT, 0);
+  EXPECT_EQ(1, config_->stats().downstream_sess_total_.value());
+  EXPECT_EQ(1, config_->stats().downstream_sess_active_.value());
+  checkTransferStats(5 /*rx_bytes*/, 1 /*rx_datagrams*/, 0 /*tx_bytes*/, 0 /*tx_datagrams*/);
+
+  test_sessions_[0].recvDataFromUpstream("world");
+  checkTransferStats(5 /*rx_bytes*/, 1 /*rx_datagrams*/, 5 /*tx_bytes*/, 1 /*tx_datagrams*/);
+}
+
+// Make sure socket option is set correctly if use_original_src_ip is set in case of ipv6.
+TEST_F(UdpProxyFilterTestIpv6, SocketOptionForUseOriginalSrcIpInCaseOfIpv6) {
+  if (!isTransparentSocketOptionsSupported()) {
+    // The option is not supported on this platform. Just skip the test.
+    GTEST_SKIP();
+  }
+
+  InSequence s;
+
+  EXPECT_CALL(os_sys_calls_, supportsIpTransparent());
+
+  setup(R"EOF(
+stat_prefix: foo
+cluster: fake_cluster
+use_original_src_ip: true
+)EOF");
+
+  expectSessionCreate(upstream_address_v6_);
+  test_sessions_[0].expectUpstreamWriteOriginalSrcIp();
+  test_sessions_[0].expectUpstreamWrite("hello", 0, peer_address_->ip());
+  recvDataFromDownstream(peer_address_->asString(), "[2001:db8:85a3::9a2e:370:7335]:80", "hello");
+
+  checkSocketOptions(test_sessions_[0], ENVOY_SOCKET_IP_TRANSPARENT, 0,
+                     ENVOY_SOCKET_IPV6_TRANSPARENT, 1);
+  EXPECT_EQ(1, config_->stats().downstream_sess_total_.value());
+  EXPECT_EQ(1, config_->stats().downstream_sess_active_.value());
+  checkTransferStats(5 /*rx_bytes*/, 1 /*rx_datagrams*/, 0 /*tx_bytes*/, 0 /*tx_datagrams*/);
+
+  test_sessions_[0].recvDataFromUpstream("world");
+  checkTransferStats(5 /*rx_bytes*/, 1 /*rx_datagrams*/, 5 /*tx_bytes*/, 1 /*tx_datagrams*/);
+}
+
+// Make sure socket option is not set correctly if use_original_src_ip is not set.
+TEST_F(UdpProxyFilterTestIpv4Ipv6, NoSocketOptionForDontUseOriginalSrcIp) {
```

Comment:
The test comment and test name here and below is not clear enough to tell what's the difference.

Reply:
My intention is that the default value of use_original_src_ip should be false if it does not exist.
I will clarify the comments and test names.